Fixes #2781

stuartbreckenridge opened the pull request:
Updating the toolbar in response to a UserDefault change notification caused the main thread warnings.

brentsimmons commented:
What code was it that changed a UserDefaults value outside the main thread? That’s not necessarily a bad thing, but it does raise a red flag.

stuartbreckenridge commented:
I am not convinced that it is NNW code that is changing the user defaults outside the main thread. 

In each one of our `set { }` blocks in `AppDefaults` I added `assert(Thread.isMainThread)` and they were never triggered, but the `assert(Thread.isMainThread)` I added in the function in the PR was triggered. I also tried using `dump()` on the notification itself, but it didn't reveal anything useful.

brentsimmons commented:
Cool — thanks for checking!
